src/main/java/org/apache/mesos/hdfs/state/PersistentState.java

```diff
@@ -149,6 +201,8 @@ public void removeTaskId(String taskId) {
         if (entry.getValue() != null && entry.getValue().equals(taskId)) {
           journalNodes.put(entry.getKey(), null);
           setJournalNodes(journalNodes);
+          Date date = DateUtils.addSeconds(new Date(), conf.getDeadNodeTimeout());
+          deadJournalNodeTimeStamp = new Timestamp(date.getTime());
```

Comment:
Doesn't have to be the same host that keeps failing/recovering. It could be a rolling upgrade/restart of all DNs in the cluster, except one of them fails to recover. Imagine the scenario where you want to actually update the Hadoop/executor binaries across each of the nodes, or a cluster-wide OS/hardware upgrade. But I suppose an operator might be okay with waiting for the entire upgrade process to complete before removing the dead node(s), although they shouldn't have to.

I will still argue that the "proper" solution would be a timeout per host, per node-type, but what you have now is good enough to merge in as is, as long as we can track my proposed solution in a TODO/issue somewhere.